src/vm/exceptionhandling.cpp

```diff
@@ -6184,11 +6187,11 @@ bool ExceptionTracker::IsInStackRegionUnwoundBySpecifiedException(CrawlFrame * p
 
     // Remember that sfLowerBound and sfUpperBound are in the "OS format".
     // Refer to the comment for CallerStackFrame for more information.
-#if defined(_TARGET_AMD64_)
+#ifndef USE_FUNCLET_CALL_HELPER
```

Comment:
Based on the comments above, this check differs based on whether the bounds are initialized from the caller SP or the SP of the frame itself. So it seems to me that using USE_FUNCLET_CALL_HELPER is a bit misleading. While it is true now that the usage of call helper matches the difference between using caller SP or SP of the frame itself, these things are not necessarily related. 
So I would prefer keeping the bounds checks here and below around likes 6282 and 6308 conditionalized by the target.